### PR TITLE
#8 GitHub actions

### PR DESCRIPTION
**Pull request by abogushov**

### CODE REVIEW

**Comment by abogushov** [comment]
@aamalev there is some deadlock in test_sa_connector.py. All tests passed locally.

**Comment by abogushov** [comment]
I managed this problem. I run local tests using asyncpg==0.21.0 but the new release came https://pypi.org/project/asyncpg/0.22.0/ a few hours ago and broke the pipeline.

**Comment by abogushov** [comment]
- I use poetry-dynamic-versioning for managing versions via git tags.

- There are no tests on Windows and Mac OS because Github Actions does not support Postgres on this platforms.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Build
+on:
+  push:
+    tags:
+      - v*
```
Comment (aamalev) [inline .github/workflows/build.yaml:5]: do you need a prefix? Traditionally we didn't have a prefix. And for example, Poetry himself does not use the prefix

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Install poetry
+        run: pip install poetry poetry-dynamic-versioning
+      - name: Set version
+        run: poetry-dynamic-versioning
+      - name: Build library
+        run: poetry build
+      - name: Publish
+        shell: bash
+        env:
+          PYPI_USERNAME: ${{ secrets.PYPI_USERNAME }}
```
Comment (aamalev) [inline .github/workflows/build.yaml:24]: Maybe it's better to use a __token__?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (abogushov) [reply .github/workflows/build.yaml:5]: This is the special prefix for poetry-dynamic-versioning used to identify version tags. The prefix will not be used in versioning.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (abogushov) [reply .github/workflows/build.yaml]: Done

File: .github/workflows/build.yaml (hunk 1 above)
Comment (abogushov) [reply .github/workflows/build.yaml:5]: It is possible to set a custom pattern if it matters.

Comment (aamalev) [reply .github/workflows/build.yaml:5]: may be set?

```
[tool.poetry-dynamic-versioning]
pattern = "(?x)^(?P<base>\\d+\\.\\d+\\.\\d+)(-?((?P<stage>[a-zA-Z]+)\\.?(?P<revision>\\d+)?))?(\\+(?P<tagged_metadata>.+))?$"
```

**Review by aamalev — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (abogushov) [reply .github/workflows/build.yaml]: Done.

**Pull request merged**